src/SignalR/server/Core/src/SerializedHubMessage.cs

```diff
@@ -63,42 +63,45 @@ public ReadOnlyMemory<byte> GetSerializedMessage(IHubProtocol protocol)
                     }
 
                     serialized = protocol.GetMessageBytes(Message);
-                    SetCache(protocol.Name, serialized);
+                    SetCacheUnsynchronized(protocol.Name, serialized);
                 }
 
                 return serialized;
             }
         }
 
-        // Used for unit testing.
-        internal IReadOnlyList<SerializedMessage> GetAllSerializations()
+        /// <summary>
+        /// Gets all serialized hub messages for all protocols used on this <see cref="SerializedHubMessage"/> instance.
+        /// </summary>
+        /// <returns>An <see cref="IEnumerable{T}"/> of already serialized hub messages for each protocol.</returns>
+        public IEnumerable<SerializedMessage> GetAllSerializations()
```

Comment:
I think this should be IList instead, everywhere we use this we immediately ToList it.